server/src/main/java/org/apache/iotdb/db/service/DataNode.java

```diff
@@ -132,200 +142,304 @@ public static void main(String[] args) {
-      // register current DataNode to ConfigNode
-      registerInConfigNode();
-      // active DataNode
+      if (isFirstStart) {
+        // Register this DataNode to the cluster when first start
+        sendRegisterRequestToConfigNode();
+      } else {
+        // Send restart request of this DataNode
+        sendRestartRequestToConfigNode();
+      }
+
+      // Active DataNode
       active();
-      // setup rpc service
+
+      // Setup rpc service
       setUpRPCService();
-      registerManager.register(MetricService.getInstance());
-
-      // init metric service
-      if (MetricConfigDescriptor.getInstance()
-          .getMetricConfig()
-          .getInternalReportType()
-          .equals(InternalReporterType.IOTDB)) {
-        MetricService.getInstance().updateInternalReporter(new IoTDBInternalReporter());
-      }
-      MetricService.getInstance().startInternalReporter();
-      // bind predefined metrics
-      DataNodeMetricsHelper.bind();
+
+      // Setup metric service
+      setUpMetricService();
 
       logger.info("IoTDB configuration: " + config.getConfigMessage());
       logger.info("Congratulation, IoTDB DataNode is set up successfully. Now, enjoy yourself!");
-    } catch (StartupException e) {
+
+    } catch (StartupException | ConfigurationException | IOException e) {
       logger.error("Fail to start server", e);
+      if (isFirstStart) {
+        // Delete the system.properties file when first start failed.
+        // Therefore, the next time this DataNode is start will still be seen as the first time.
+        SYSTEM_PROPERTIES.deleteOnExit();
+      }
       stop();
     }
   }
 
-  /** initialize the current node and its services */
-  public boolean initLocalEngines() {
-    return true;
-  }
-
   /** Prepare cluster IoTDB-DataNode */
-  private void prepareDataNode() throws StartupException {
-    // check iotdb server first
+  private boolean prepareDataNode() throws StartupException, ConfigurationException, IOException {
+    // Set cluster mode
+    config.setClusterMode(true);
+
+    // Notice: Consider this DataNode as first start if the system.properties file doesn't exist
+    boolean isFirstStart = !SYSTEM_PROPERTIES.exists();
+
+    // Check target ConfigNodes
+    for (TEndPoint endPoint : config.getTargetConfigNodeList()) {
+      if (endPoint.getIp().equals("0.0.0.0")) {
+        throw new StartupException(
+            "The ip address of any target_config_node_list couldn't be 0.0.0.0");
+      }
+    }
+
+    // Set this node
+    thisNode.setIp(config.getInternalAddress());
+    thisNode.setPort(config.getInternalPort());
+
+    // Startup checks
     StartupChecks checks = new StartupChecks(IoTDBConstant.DN_ROLE).withDefaultTest();
     checks.verify();
 
-    // Register services
-    JMXService.registerMBean(getInstance(), mbeanName);
+    // Check system configurations
+    IoTDBStartCheck.getInstance().checkSystemConfig();
+
+    return isFirstStart;
```

Comment:
return !SYSTEM_PROPERTIES.exists();

Reply:
No. The system.properties file will be created in IoTDBStartCheck.getInstance().checkSystemConfig(). So we should get the isFirstStart at the beginning